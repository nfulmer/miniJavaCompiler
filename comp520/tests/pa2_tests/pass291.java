// PA2 pass references in expresssions 
class Refs {

    void p() {
    	x1 = a;
    	x2 = a.b;
        x3 = this;
        x4 = a[n];
        x5 = f(n);
        x6 = 6;
        x7 = false;
        x8 = new Foo();
        x9 = new Foo[n];
        x10 = new int [n];
    }
}
