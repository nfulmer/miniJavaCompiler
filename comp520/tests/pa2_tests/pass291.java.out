======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Refs" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [10]
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x1" Identifier
  .   .   .   RefExpr
  .   .   .     IdRef
  .   .   .       "a" Identifier
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x2" Identifier
  .   .   .   RefExpr
  .   .   .     QualRef
  .   .   .       "b" Identifier
  .   .   .       IdRef
  .   .   .         "a" Identifier
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x3" Identifier
  .   .   .   RefExpr
  .   .   .     ThisRef
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x4" Identifier
  .   .   .   IxExpr
  .   .   .     IdRef
  .   .   .       "a" Identifier
  .   .   .     RefExpr
  .   .   .       IdRef
  .   .   .         "n" Identifier
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x5" Identifier
  .   .   .   CallExpr
  .   .   .     IdRef
  .   .   .       "f" Identifier
  .   .   .     ExprList + [1]
  .   .   .     . RefExpr
  .   .   .     .   IdRef
  .   .   .     .     "n" Identifier
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x6" Identifier
  .   .   .   LiteralExpr
  .   .   .     "6" IntLiteral
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x7" Identifier
  .   .   .   LiteralExpr
  .   .   .     "false" BooleanLiteral
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x8" Identifier
  .   .   .   NewObjectExpr
  .   .   .     ClassType
  .   .   .       "Foo" Identifier
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x9" Identifier
  .   .   .   NewArrayExpr
  .   .   .     ClassType
  .   .   .       "Foo" Identifier
  .   .   .     RefExpr
  .   .   .       IdRef
  .   .   .         "n" Identifier
  .   .   . AssignStmt
  .   .   .   IdRef
  .   .   .     "x10" Identifier
  .   .   .   NewArrayExpr
  .   .   .     INT BaseType
  .   .   .     RefExpr
  .   .   .       IdRef
  .   .   .         "n" Identifier
=============================================
